mysql-test/suite/rpl/t/rpl_mts_spco_deadlock_slave_trans_retries_assertion.test

```diff
@@ -0,0 +1,159 @@
+# ==== Purpose ====
+#
+# This test verifies that deadlock errors arising due to lock conflicts are
+# handled properly when --slave-preserve-commit-order is enabled on the replica
+# server.
+#
+# ==== Implementation ====
+#
+# 1. Create source-replica topolgy.
+# 2. Setup testing environment.
+#    2.1. Create table on source and insert few rows.
+#    2.2. Setup necessary variables on replica server.
+# 3. Execute transactions on source server that would generate deadlock on the
+#    replica server.
+#
+#    The commands used to generate deadlock in this test are:
+#
+#    INSERT INTO t1 (11, NULL);
+#    DELETE FROM t1 WHERE c2 <= 3;
+#
+#    With the help of debug points on the source server, simulate that they are
+#    executed parallely on master and make sure INSERT is binlogged before
+#    DELETE.
+#
+#    On replica server, they are applied parallel by w1 and w2. And the deadlock is:
+#      w1                                        w2
+#    -------------------------                 --------------------------
+#     applying INSERT                          applying DELETE
+#                                              get the row locks first.
+#     waiting for w1 to release
+#     the row lock.
+#                                              waiting for w2 to commit
+#
```

Comment:
in this example, why w1 would wait for itself? is it mistake  ? s/w1/w2?

similary on the otherside w2 is waiitng for itself to commit? is it w1? "waitiing for w1 to commit"?

Reply:
This is a type error. It should have been  
```
#    On replica server, they are applied parallel by w1 and w2. And the deadlock is:
#      w1                                        w2
#    -------------------------                 --------------------------
#                                              get the row locks first.
#     waiting for w2 to release
#     the row lock.
#                                              waiting for w1 to commit
#
```

Reply:
This is a type error. It should have been  
```
#    On replica server, they are applied parallel by w1 and w2. And the deadlock is:
#      w1                                        w2
#    -------------------------                 --------------------------
#                                              get the row locks first.
#     waiting for w2 to release
#     the row lock.
#                                              waiting for w1 to commit
#
```